compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2103,7 +2103,11 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             }
         }
 
-        err("`?Trait` bounds are only permitted at the point where a type parameter is declared")
+        err("this relaxed bound is not permitted here")
+            .with_note(
+                "in this context, relaxed bounds are only allowed on \
```

Comment:
"in this context" because in other contexts like in assoc tys, relaxed bounds are also allowed in other places (in that case: assoc ty item bounds).